src/containers/RedashContainer/index.jsx

```diff
@@ -2,9 +2,55 @@ import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { withStyles } from '@material-ui/core/styles';
 import ChartJsWrapper from '../../components/ChartJsWrapper';
-import telemetryDataToDatasets from '../../utils/chartJs/redashFormatter';
 import fetchJson from '../../utils/fetchJson';
 import { withErrorBoundary } from '../../vendor/errors';
+import generateDatasetStyle from '../../utils/chartJs/generateDatasetStyle';
+import SETTINGS from '../../settings';
+import { Log } from '../../vendor/logs';
+
+/* eslint-disable camelcase */
+
+const sortByDate = (a, b) =>
+  new Date(b.submission_date) - new Date(a.submission_date);
+const dataToChartJSformat = data =>
+  data.map(({ submission_date, value }) => ({
+    x: submission_date,
+    y: value,
+  }));
+const telemetryDataToDatasets = (data, dataKeyIdentifier) => {
+  const queryResulset = data.query_result;
+  // Separate data points into buckets
+  const buckets = queryResulset.data.rows.reduce((result, datum) => {
+    const key = datum[dataKeyIdentifier];
+
+    if (!key) {
+      // XXX: Make it a custom error, catch it and display a UI message
+      Log.error(
+        'Check the Redash data and determine what is the key used to categorize the data.'
+      );
+    }
+
+    if (!result[key]) {
+      // eslint-disable-next-line no-param-reassign
+      result[key] = [];
+    }
+
+    result[key].push(datum);
+
+    return result;
+  }, {});
+  const datasets = Object.keys(buckets).map((key, index) => {
+    const datum = buckets[key];
+
+    return {
+      label: key,
+      ...generateDatasetStyle(SETTINGS.colors[index]),
+      data: dataToChartJSformat(datum.sort(sortByDate)),
+    };
+  });
+
+  return { datasets };
+};
```

Comment:
the above is from the file  https://github.com/mozilla-frontend-infra/firefox-health-dashboard/pull/414/files#diff-729a1e388df9d305582838f2b94f0caf